app/controllers/supplementary/SummaryPageController.scala

```diff
@@ -51,11 +53,16 @@ class SummaryPageController @Inject()(
 
   def displayPage(): Action[AnyContent] = authenticate.async { implicit request =>
     customsCacheService.fetch(supplementaryCacheId).map {
-      case Some(cacheMap) => Ok(summary_page(appConfig, SupplementaryDeclarationData(cacheMap)))
-      case None           => Ok(summary_page_no_data(appConfig))
+      case Some(cacheMap) if containsMandatoryData(cacheMap) =>
+        Ok(summary_page(appConfig, SupplementaryDeclarationData(cacheMap)))
+      case Some(_) => Ok(summary_page_no_data(appConfig))
+      case None    => Ok(summary_page_no_data(appConfig))
```

Comment:
Instead of two cases you can use one with `_`.

We start moving appConfig as implicit one, so if you want you can move it too, using views are shorter :)

Reply:
I've changed both.